### ISSUE TITLE
classify_enable_learning does not work when using multiple languages

### ISSUE THREAD

**Issue by lorenzob** [tesseract-ocr/tesseract#1537]
### Environment

* **Tesseract Version**:

tesseract 3.05.01
 leptonica-1.75.3
  libjpeg 8d (libjpeg-turbo 1.4.2) : libpng 1.6.20 : zlib 1.2.8

* **Commit Number**: (built from HEAD a few days ago, I do not know the exact commit number)

* **Platform**: Linux  4.13.0-39-generic #44~16.04.1-Ubuntu SMP Thu Apr 5 16:43:10 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

### Current Behavior:

Using the api the variable _classify_enable_learning_ correctly disables the learning and if I call the GetUTF8Text multiple times on the same text I get the same result.

If I use multiple languages in the Init() call (ex. "eng+ita") it looks like the _classify_enable_learning_ variable is ignored and the confidence changes with multiple calls.

I'm setting _classify_enable_learning_ and _classify_enable_adaptive_matcher_ both to 0.

I do not know if this is expected behaviour or not and if there is a way to disable learning also when using multiple languages.

This is a followup to this issue: https://github.com/tesseract-ocr/tesseract/issues/1535 please look there for sample code and images.

**Comment by amitdo** [comment]
Try this:
https://github.com/tesseract-ocr/tesseract/blob/e95ff1159e652d9b8ae6bc4aafdb196981942e6a/src/api/baseapi.h#L326

**Comment by amitdo** [comment]
@lorenzob, did you try what I suggested?

**Comment by lorenzob** [comment]
Yes, I tried. With my local installation is completely ignored like the other parameters. With another one the classify_enable_learning flags are enough to disable learning.
I suppose my installation (built locally with latest tesseract and leptonica) is somehow broken and I stopped using it.
Please let me know if and how we can investigate this further otherwise feel free to close this issue.

**Comment by amitdo** [comment]
I ask again, did you try calling in your code **ClearAdaptiveClassifier()** after each page recognition?

Your answer mentioned just setting parameters.

**Comment by lorenzob** [comment]
Please read again my answer: yes, I tried.

**Comment by amitdo** [comment]
>yes, I tried.

I saw it, but was not sure to what it referred.
Now I'm sure :-)

I have no other option to suggest.

**Comment by lorenzob** [comment]
I tried (to use 4.0.0-beta from master :) ).

The main reason I'm not using it is that it's missing support for whitelist so in practice I get better results with the 3.05.

I also discovered that adaptive learning does improve the results, so I'd like to use it but in a more controlled way. I asked for suggestions here:

https://groups.google.com/forum/#!topic/tesseract-ocr/sXH3FMRlJ6k

but didn't get many answers.

I'd like to manually set the "best learned" parameters, if possible. Or just find a set of parameters to give me results close to the one I get with adaptive learning. My samples are very simple and uniform, I'm sure there are many options for fine tuning.
